binaries/daemon/src/spawn.rs

```diff
@@ -148,20 +153,59 @@ pub async fn spawn_node(
                 }
             }
 
-            command.spawn().wrap_err(format!(
-                "failed to run runtime {}/{}",
-                runtime_config.node.dataflow_id, runtime_config.node.node_id
-            ))?
+            command
+                .stdin(Stdio::null())
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped())
+                .spawn()
+                .wrap_err(format!(
+                    "failed to run runtime {}/{}",
+                    runtime_config.node.dataflow_id, runtime_config.node.node_id
+                ))?
         }
     };
 
     tokio::spawn(async move {
+        // let hlc = HLC::default();
+        // let timestamp = hlc.new_timestamp().to_string();
+        // let time = timestamp
+        // .split('.')
+        // .next()
+        // .expect("Could not extract date from timestamp."); // TODO: Add time within log file name
+
+        let mut file = File::create(format!("logs/{node_id}.txt"))
+            .await
+            .expect("Failed to create log file");
+
+        let mut stdout_lines =
+            (tokio::io::BufReader::new(child.stdout.take().expect("failed to take stdout")))
+                .lines();
+
+        while let Ok(Some(line)) = stdout_lines.next_line().await {
+            file.write(line.as_bytes()).await.unwrap();
+            file.write(b"\n").await.unwrap();
+            info!(line);
+        }
+
+        let mut stderr_lines =
+            (tokio::io::BufReader::new(child.stderr.take().expect("failed to take stderr")))
+                .lines();
```

Comment:
Reading the stderr only after the stdout was closed can lead to deadlocks because the `stderr` pipe might be full and block the sender (i.e. the spawned node).

Reply:
Agreed. So, I have replaced this by spawning a `stdout` thread and a `stderr` thread that can push string to a file writer thread.